packages/wdio-browserstack-service/src/service.ts

```diff
@@ -0,0 +1,240 @@
+import logger from '@wdio/logger'
+import got, { GotRequestFunction } from 'got'
+import WebDriver from 'webdriver'
+
+import { getBrowserDescription, getBrowserCapabilities, isBrowserstackCapability } from './util'
+
+const log = logger('@wdio/browserstack-service')
+
+type Capabilities = WebDriver.Capabilities & WebdriverIO.MultiRemoteCapabilities;
+
+type Browser = WebdriverIO.BrowserObject & WebdriverIO.MultiRemoteBrowserObject;
+
+type Context = any;
+
+type Feature = {
+    document: {
+        feature: {
+            name: string;
+        }
+    }
+};
+
+type Pickle = {
+    name: string;
+}
+
+type MultiRemoteAction = (sessionId: string, browserName?: string) => ReturnType<GotRequestFunction> | Promise<void>;
+
+type SessionResponse = {
+    // eslint-disable-next-line camelcase
+    automation_session: {
+        // eslint-disable-next-line camelcase
+        browser_url: string
+    }
+}
+
+export default class BrowserstackService implements WebdriverIO.ServiceInstance {
+    private _sessionBaseUrl: string = 'https://api.browserstack.com/automate/sessions';
+    private _failReasons: string[] = [];
+    private _scenariosThatRan: string[] = [];
+    private _preferScenarioName: boolean;
+    private _strict: boolean;
+    private _failureStatuses: string[] = ['failed', 'ambiguous', 'undefined', 'unknown'];
+    private _caps: Capabilities;
+    private _browser?: Browser;
+    private _fullTitle?: string;
+    constructor (options: BrowserstackConfig = {}, caps: Capabilities, private _config: WebdriverIO.Config) {
+        // Cucumber specific
+        this._preferScenarioName = Boolean(options.preferScenarioName)
+        this._strict = Boolean(_config.cucumberOpts && _config.cucumberOpts.strict)
+        // See https://github.com/cucumber/cucumber-js/blob/master/src/runtime/index.ts#L136
+        this._strict && this._failureStatuses.push('pending')
+        this._caps = caps
+    }
```

Comment:
some improvements here:
```suggestion
    constructor (private _options: BrowserstackConfig = {}, private _caps: Capabilities, private _config: WebdriverIO.Config) {
        if (this._strict) {
            this._failureStatuses.push('pending')
        }
    }
```

`_strict` is nowhere else used so we can get rid of it and by making the whole options object a class property we don't need to specifically define `_preferScenarioName`. The comment pointed to line of code that doesn't really indicate any information as to why this was implemented like it.

Reply:
Removing these will break some tests:
```ts
this._preferScenarioName = Boolean(_options.preferScenarioName)
this._strict = Boolean(_config.cucumberOpts && _config.cucumberOpts.strict)
```

Reply:
the tests should be updated then.

Reply:
Did some changes, but I still have a question, follow up in this comment: https://github.com/webdriverio/webdriverio/pull/6172#discussion_r553558912